galley/pkg/config/analysis/analyzers/annotations/annotations.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"strings"
+
+	"istio.io/api/annotation"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/galley/pkg/config/meta/metadata"
+	"istio.io/istio/galley/pkg/config/meta/schema/collection"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// K8sAnalyzer checks for misplayed and invalid Istio annotations in K8s resources
+type K8sAnalyzer struct{}
+
+var (
+	// Currently we don't have an Istio API that tells which Istio annotations apply
+	// to which K8s and custom resources.  This map is currently hand-crafted, but
+	// could be generated similarly to https://istio.io/docs/reference/config/annotations/
+	annotationAppliesTo = map[*annotation.Instance][]string{
```

Comment:
FYI @geeknoid might it make sense to merge annotation tools?

Reply:
We're already covered...

This code should be pulling in the generated file in istio/api/annotations/annotations.gen.go. This includes the full current set of annotations. Sadly, it doesn't include the newly introduced resource type data that this analysis needs. I'll get this fixed, should be rolled out within a day.

Reply:
https://github.com/istio/tools/pull/426

Reply:
Once https://github.com/istio/api/pull/1121 merges, this PR can pull in the new .gen.go file to get all the necessary definitions.